### PR TITLE
docs: Add NavigationBar docs

### PR DESCRIPTION
**Pull request by kazo0**
## PR Type

What kind of change does this PR introduce?
- Documentation content changes

### CODE REVIEW

File: doc/controls/NavigationBar.md [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+
+#### iOS
+![](../assets/navbar-ios-page1.png)
+
+![](../assets/navbar-ios-page2.png)
+
+| Platform | Native control      | Benefits                                              |
+|----------|---------------------|-------------------------------------------------------|
+| Android  | `Toolbar`          | Native pressed states (ripple), native overflow menu. |
+| iOS      | `UINavigationBar` | Transitions when navigating  between pages.   |
```
Comment (agneszitte-nventive) [inline doc/controls/NavigationBar.md:48]:
```suggestion
| iOS      | `UINavigationBar` | Transitions when navigating between pages.   |
```

File: doc/controls/NavigationBar.md [hunk 2]
```diff
@@ -0,0 +1,643 @@
...
+
+[Source (Android)](https://material.io/guidelines/layout/structure.html#structure-app-bar)
+
+
+## MainCommandMode
+The `NavigationBar` has a property named `MainCommandMode` that can be set to either:
+1. `MainCommandMode.Back` (default)
+1. `MainCommandMode.Action`
+
+`MainCommandMode` should be set to `Action` when the `MainCommand` is being used for anything other than backward navigation, such as displaying a burger menu or dispalying a prompt to the user before the navigation occurs.
```
Comment (agneszitte-nventive) [inline doc/controls/NavigationBar.md:178]:
```suggestion
`MainCommandMode` should be set to `Action` when the `MainCommand` is being used for anything other than backward navigation, such as displaying a burger menu or displaying a prompt to the user before the navigation occurs.
```

File: doc/controls/NavigationBar.md [hunk 3]
```diff
@@ -0,0 +1,643 @@
...
+* The `NavigationBar` must stretch horizontally and be aligned with the top of your `Page`
+* The `NavigationBar` can't move (i.e., don't put it inside a `ScrollViewer`)
+* The `NavigationBar` must be accessible as soon as the `Page` is being navigated to (i.e., don't put it inside a `DataTemplate`)
+* There can only be one `NavigationBar` per `Page`
+
+## Extensibility
+
+The `NavigationBar` is automatically managed by the `Frame` control, however you can still use the "native" mode of the `NavigationBar` with your own navigation mechanism.
+
+On **iOS** a `NavigationBarHelper` is available for this purpose, you only have to invoke each of the provided method in your own `UIViewController` implementation.
```
Comment (agneszitte-nventive) [inline doc/controls/NavigationBar.md:245]:
```suggestion
On **iOS** a `NavigationBarHelper` is available for this purpose, you only have to invoke each of the provided methods in your own `UIViewController` implementation.
```

File: doc/controls/NavigationBar.md [hunk 4]
```diff
@@ -0,0 +1,792 @@
...
+* The `NavigationBar` must be accessible as soon as the `Page` is being navigated to (i.e., don't put it inside a `DataTemplate`)
+* There can only be one `NavigationBar` per `Page`
+
+## Extensibility
+
+The `NavigationBar` is automatically managed by the `Frame` control, however you can still use the "native" mode of the `NavigationBar` with your own navigation mechanism.
+
+On **iOS** a `NavigationBarHelper` is available for this purpose, you only have to invoke each of the provided methods in your own `UIViewController` implementation.
+
+## AppBarButton
```
Comment (Xiaoy312) [inline doc/controls/NavigationBar.md:396]: could be bumped to a tier-1 #header
adjust child nodes accordingly if taken

File: doc/controls/NavigationBar.md [hunk 5]
```diff
@@ -0,0 +1,792 @@
...
+
+When `AppBarButton` is used within a native `NavigationBar`, its control template is completely ignored and can't be customized.
+
+### Events
+
+| Event   | Windows | iOS | Android | Comments |
+|---------|:-------:|:---:|:-------:|----------|
+| Clicked | x       | x   | x       |          |
+
+### Properties
```
Comment (Xiaoy312) [inline doc/controls/NavigationBar.md:408]: the properties headers should be a tier lower and this header; they are currently on same tier.

File: doc/controls/NavigationBar.md [hunk 6]
```diff
@@ -0,0 +1,792 @@
...
+
+```cs
+<Style TargetType="NavigationBar" BasedOn="{StaticResource NativeDefaultNavigationBar}" />
+```
+
+#### Remarks
+
+In this mode, the `NavigationBar` can't be fully customized like other templatable controls would. Additionally, you can't customize the visual states of either the `NavigationBar` or its `AppBarButtons`.
+
+### Padding
```
Comment (Xiaoy312) [inline doc/controls/NavigationBar.md:62]: should be a tier-4 ####header under "### Native", no?

File: doc/controls/NavigationBar.md [hunk 7]
```diff
@@ -0,0 +1,792 @@
...
+
+By default on iOS and Android, the `VisibleBoundsPadding.PaddingMask` on `NavigationBar` is set to `"Top"` to properly support the screen notch or punch-holes.
```
Comment (Xiaoy312) [inline doc/controls/NavigationBar.md:64]:
```suggestion
By default on iOS and Android, the `VisibleBoundsPadding.PaddingMask` on `NavigationBar` is set to `Top` to properly support the screen notch or punch-holes.
```

**Review by Xiaoy312 — COMMENTED** [review]
- update this line: https://github.com/unoplatform/uno.toolkit.ui/blob/f495d77996012331653b13bbfb953e67a61be43c/doc/controls-styles.md?plain=1#L8
- maybe consider remove the marker header tag for "remarks", it adds another level to the hierarchy without much to gain? "remarks:" would probably be suffice.

**Review by Xiaoy312 — APPROVED** [review]

**Pull request merged**